src/Components/SmartComponents/SystemsPage/SystemsPage.js

```diff
@@ -95,22 +96,39 @@ const SystemsPage = () => {
     };
 
     const doOptOut = (systemId = null, optOut) => {
+        let count = systemId ? 1 : selectedHosts.length || 0;
+        let systemName;
+
+        if (count === 1) {
+            systemName = systems.find(system => system.id === (systemId ? systemId[0] : selectedHosts[0]))?.display_name;
```

Comment:
is `systemId` really an array here? :thinking:

Reply:
Yes, to make it compatible with array when multiple are optOut, one more opportunity to consider TypeScript.